src/Provider/Node/SystemNodeProvider.php

```diff
@@ -73,4 +78,35 @@ protected function getIfconfig()
 
         return ob_get_clean();
     }
+
+    /**
+     * Returns mac address from the first system interface via the sysfs interface
+     *
+     * @return string|bool
+     */
+    protected function getsysfs()
+    {
+        $mac = false;
+        if (strtoupper(php_uname('s')) === "LINUX") {
+            // get all the macadresses of all systems
+            $macs = array_map(
+                'file_get_contents',
+                glob('/sys/class/net/*/address', GLOB_NOSORT)
+            );
+
+            $macs = array_map('trim', $macs);
+
+            // remove invalid entries
+            $macs = array_filter($macs, function ($mac) {
+                return
+                    // localhost adapter
+                    $mac !== '00:00:00:00:00:00' &&
+                    // must match  mac adress
+                    preg_match('/^([0-9a-f]{2}:){5}[0-9a-f]{2}$/i', $mac);
```

Comment:
Should this use the same pattern that is used in `getNode()` earlier this in this class? If so, I'd move that pattern into a class constant and use it in both places, to avoid duplication.

Reply:
no i think we should not, this patterns has an anchor to the start and the end of the line, while the other patterns requires a char, not being an `:`  on both sides of the macaddress. While both patterns look alike, the are both for different purposes and expected input, and thus subtle different.